CRM/Utils/Hook.php

```diff
@@ -2835,4 +2835,23 @@ public static function alterApiRoutePermissions(&$permissions, $entity, $action)
     );
   }
 
+  /**
+   * Allows an extension to override the checksum validation.
+   * For example you may want to invalidate checksums that were sent out/forwarded by mistake. You could also
+   * intercept and redirect to a different page in this case - eg. to say "sorry, you tried to use a compromised
+   * checksum".
+   *
+   * @param int $contactID
+   * @param string $checksum
+   * @param bool $invalid
+   *   Leave this at FALSE to allow the core code to perform validation. Set to TRUE to invalidate
+   */
+  public static function invalidateChecksum($contactID, $checksum, &$invalid) {
+    return self::singleton()->invoke(
+      ['contactID', 'checksum', 'invalid'],
+      $contactID, $checksum, $isValid, self::$_nullObject, self::$_nullObject,
```

Comment:
`$isValid` - maybe this was left over from the previous version - should be `$invalid`

Reply:
Fixed!